configs/agent_config.go

```diff
@@ -53,6 +55,72 @@ type AgentHooks struct {
 	DoOnWorkflowEnd   string `yaml:"do_on_workflow_end"`
 }
 
+func RegisterAgentOverrides() error {
```

Comment:
I had to closely look at and compare all of 4 sections of the dir handling sections inside this function to discover that they are exactly the same. This made me wonder if it would be good to group them and create a single logic with multiple input parameters for them to indicate that they are the same. 

Something like
```
params := []struct {
	name   string
	dir    string
	envKey string
}{
	{
		name:   "BITRISE_DATA_HOME_DIR",
		dir:    config.BitriseDirs.BitriseDataHomeDir,
		envKey: BitriseDataHomeDirEnvKey,
	},
	{
		name:   "BITRISE_SOURCE_DIR",
		dir:    config.BitriseDirs.SourceDir,
		envKey: BitriseSourceDirEnvKey,
	},
	{
		name:   "BITRISE_DEPLOY_DIR",
		dir:    config.BitriseDirs.DeployDir,
		envKey: BitriseDeployDirEnvKey,
	},
	{
		name:   "BITRISE_TEST_DEPLOY_DIR",
		dir:    config.BitriseDirs.TestDeployDir,
		envKey: BitriseTestDeployDirEnvKey,
	},
}
for _, param := range params {
	err = pathutil.EnsureDirExist(param.dir)
	if err != nil {
		return fmt.Errorf("can't create %s: %w", param.name, err)
	}
	err = os.Setenv(param.envKey, param.dir)
	if err != nil {
		return fmt.Errorf("set %s: %w", param.name, err)
	}
}
```

Reply:
Thank you, I fully agree, please take a look now